### PR TITLE
Add use-context command

### PR DESCRIPTION
**Pull request by rafiramadhana**
## Why this matters

Resolves https://github.com/ergomake/layerform/issues/89

## Solution

Add command `use-context` to use the selected current context.

## How to test it

Run the e2e.yml.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -49,6 +49,13 @@ jobs:
           layerform config set-context test-cloud -t cloud --url https://demo.layerform.dev --email foo@bar.com --password strongpass
           layerform config set-context test-local -t local --dir test
 
+      - name: layerform config select-context
+        run: |
+          # fails if command succeeds
+          ! layerform config select-context test-does-not-exist # context does not exist
+
+          layerform config set-context test-local -t local --dir test && layerform config select-context test-local
```
Comment (rafiramadhana) [inline .github/workflows/e2e.yml:57]: I'm still wondering what is the default layerform/configurations.yml file in our CI/CD testing env.

As the mitigation to ensure that a config exists, I do `set-context` first to create a config.

Comment (vieiralucas) [reply .github/workflows/e2e.yml:57]: Well, it is the last one that run in the previous step (check line 50).
At this point, the current context is the `test-local`, your `set-context` ends up being a `no-op`.
What I think you should do to test is somehow check if the `select-context` actually works.
You can use the `config get-contexts` command alongside with `grep` for that. The `selected` context will have a `*` in it's line.
I'll make a comment with suggestions.

File: cmd/cli/select_context.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+		cfg.CurrentContext = name
+
+		err = cfg.Save()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", errors.Wrap(err, "fail to save config file"))
+			os.Exit(1)
+		}
+
+		fmt.Fprintf(os.Stdout, "Context \"%s\" selected.\n", name)
```
Comment (vieiralucas) [inline cmd/cli/select_context.go:53]: Can we change this to:
```suggestion
		fmt.Fprintf(os.Stdout, "Switched to context \"%s\".\n", name)
```

FYI I'm getting inspiration from `kubectl`

File: cmd/cli/select_context.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			fmt.Fprintf(os.Stderr, "%s\n", errors.Wrap(err, "fail to open config file"))
+			os.Exit(1)
+		}
+
+		_, ok := cfg.Contexts[name]
+		if !ok {
+			fmt.Fprintf(
+				os.Stderr,
+				"context %s does not exist\n",
```
Comment (vieiralucas) [inline cmd/cli/select_context.go:39]: Can we change this to:
```suggestion
				"no context exists with the name \"%s\".\n",
```

**Review by vieiralucas — CHANGES_REQUESTED** [review]
Hi @rafiramadhana,

Thank you so much for the contribution, overall the implementation seems correct.
I think you got confused with the name of the command, the idea was to call it `use-context`. Do you mind changing it? Also change the name of the file and stuff.

**Comment by rafiramadhana** [comment]
@vieiralucas it looks like the configure s3 pipeline is broken

```
fail to get layers backend: fail to read file: fail to load layers from s3: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
Error: Process completed with exit code 1.
```
https://github.com/ergomake/layerform/actions/runs/6229934241/job/16909208767?pr=96

could this be related to my PR changes?

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (rafiramadhana) [reply .github/workflows/e2e.yml]: updated https://github.com/ergomake/layerform/pull/96/commits/9e15b350bd653ff464027be46301bdaeb02d878e

File: cmd/cli/select_context.go (hunk 2 above)
Comment (rafiramadhana) [reply cmd/cli/select_context.go]: updated https://github.com/ergomake/layerform/pull/96/commits/978e3c6bf38d9b22e48eaffe01c2c63f6d632b03

File: cmd/cli/select_context.go (hunk 3 above)
Comment (rafiramadhana) [reply cmd/cli/select_context.go]: updated https://github.com/ergomake/layerform/pull/96/commits/978e3c6bf38d9b22e48eaffe01c2c63f6d632b03

**Comment by vieiralucas** [comment]
Hi @rafiramadhana, this happens because your changes to the tests leave a context of type `s3` as active. Then later commands will try to use `s3` as the backend but we don't have AWS setup here in the workflows.
You can just switch the ortder of the tests you added, put the `test-local` as the last one.

**Review by vieiralucas — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -49,6 +49,23 @@ jobs:
...
+          layerform config use-context test-local
+          layerform config get-contexts | tee usecontext
+          grep -E '^\*\s+test-local' usecontext
+          ! grep -E '^\*\s+test-s3' usecontext
+
+          # switch context to test-s3
+          layerform config use-context test-s3
+          layerform config get-contexts | tee usecontext
+          ! grep -E '^\*\s+test-local' usecontext
+          grep -E '^\*\s+test-s3' usecontext
```
[the remark this answers is not in the document]
Comment (rafiramadhana) [reply .github/workflows/e2e.yml]: updated https://github.com/ergomake/layerform/pull/96/commits/3f94a8381c212887f782e1b0ad2bd6e054db84b1

thanks for the pointers

**Pull request merged**